### PR TITLE
Make beta stats the default stats

### PR DESCRIPTION
**Pull request by willdurand**
Fixes https://github.com/mozilla/addons-server/issues/14807

### CODE REVIEW

File: src/olympia/devhub/tests/test_views.py [hunk 1]
```diff
@@ -155,10 +155,9 @@ def test_show_hide_statistics_and_new_version_for_disabled(self):
         assert 'Statistics' in links, ('Unexpected: %r' % links)
         assert 'New Version' in links, ('Unexpected: %r' % links)
 
-        # Disabled (user): hide statistics and new version links.
+        # Disabled (user): hide new version link.
         self.addon.update(disabled_by_user=True)
         links = self.get_action_links(self.addon.pk)
-        assert 'Statistics' not in links, ('Unexpected: %r' % links)
```
Comment (willdurand) [inline src/olympia/devhub/tests/test_views.py:161]: stats pages are available almost all the time for the developer

File: src/olympia/stats/tests/test_cron.py [hunk 2]
```diff
@@ -68,15 +59,6 @@ def test_no_addon_or_date(self, group_mock):
...
         calls = group_mock.call_args[0][0]
 
-        # There should be 3 updates, but 2 of them have a date close enough
-        # together that they'll be indexed in the same chunk, so we should have
-        # 2 calls.
-        update_counts_calls = [
-            call.args for call in calls
-            if call.task == 'olympia.stats.tasks.index_update_counts'
-        ]
-        assert len(update_counts_calls) == 2
```
Comment (willdurand) [inline src/olympia/stats/tests/test_cron.py:78]: Things like this will be fully removed in a different patch. I removed it here because tests were failing since I changed the fixtures files to only have download counts.

File: src/olympia/stats/tests/test_views.py [hunk 3]
```diff
@@ -7,41 +7,67 @@
...
 
-from waffle.testutils import override_flag
-
 from olympia import amo
 from olympia.access.models import Group, GroupUser
 from olympia.addons.models import Addon, AddonUser
-from olympia.amo.tests import (TestCase, version_factory, addon_factory,
-                               user_factory)
-from olympia.amo.urlresolvers import reverse, resolve
+from olympia.amo.tests import (
```
Comment (willdurand) [inline src/olympia/stats/tests/test_views.py:15]: I had to rewrite most of the file because test cases verifying things like csv/json exports, etc. were based on the legacy stats. I kept almost everything and configured a BigQuery mock in each test method.

File: src/olympia/stats/tests/test_views.py [hunk 4]
```diff
@@ -163,36 +186,7 @@ def test_stats_for_user_disabled_addon(self):
         self._check_it(self.public_views_gen(format='json'), 200)
 
 
-class ESStatsTest(StatsTest, amo.tests.ESTestCase):
```
Comment (willdurand) [inline src/olympia/stats/tests/test_views.py:166]: Moved/renamed below

File: src/olympia/stats/tests/test_views.py [hunk 5]
```diff
@@ -234,353 +228,428 @@ def test_addon_anonymous(self):
         self._check_it(self.public_views_gen(addon_id=5, format='json'), 403)
 
 
-class TestCSVs(ESStatsTest):
```
Comment (willdurand) [inline src/olympia/stats/tests/test_views.py:237]: Similar test cases exist in `TestExports`. I removed the class here and kept test cases that weren't duplicated.

File: src/olympia/stats/views.py [hunk 6]
```diff
@@ -162,6 +155,8 @@ def zip_overview(downloads, updates):
         start_date = downloads[0]['date']
     if updates:
         d = updates[0]['date']
+        if isinstance(d, str):
+            d = datetime.strptime(d, '%Y-%m-%d').date()
```
Comment (eviljeff) [inline src/olympia/stats/views.py:159]: is this a drive-by fix for something else?

File: src/olympia/stats/views.py [hunk 7]
```diff
@@ -170,7 +165,10 @@ def iterator(series):
             item = next(series)
             next_date = start_date
             while True:
-                if item['date'] == next_date:
+                date = item['date']
+                if isinstance(date, str):
+                    date = datetime.strptime(date, '%Y-%m-%d').date()
```
Comment (eviljeff) [inline src/olympia/stats/views.py:170]: same here.  We're now converting string to date... but only when it's a str (what other types can it be?)

File: src/olympia/stats/views.py (hunk 6 above)
Comment (willdurand) [reply src/olympia/stats/views.py:159]: d'oh. I meant to comment on this line too, sorry. Yes. That's because BigQuery returns datetime instances but the test suite uses dates as strings. I feel like it is safer to have that. WDYT?

Comment (eviljeff) [reply src/olympia/stats/views.py:159]: I think we should be testing with the same format really, so the test data is datetime (at least by the time it's consumed by the view).  If this is going to be a big change feel free to put it in a follow-up.

Comment (willdurand) [reply src/olympia/stats/views.py]: You're right. I updated the test file with `date` instances. I also updated the BigQuery function to format series so that it returns `date` instances too. It should be the case but better safe than sorry.

File: src/olympia/stats/utils.py [hunk 8]
```diff
@@ -21,10 +23,15 @@ def rows_to_series(rows, filter_by=None):
     """Transforms BigQuery rows into series items suitable for the rest of the
     AMO stats logic."""
     for row in rows:
+        submission_date = (
+            datetime.strptime(row['submission_date'], '%Y-%m-%d').date()
+            if not isinstance(row['submission_date'], date)
+            else row['submission_date']
```
Comment (eviljeff) [inline src/olympia/stats/utils.py:29]: I thought you said BigQuery returned `date` instances already?  When are they not `date`s?  (Should there be try... except handling if the data format is incorrect?)

Comment (willdurand) [reply src/olympia/stats/utils.py]: right, although this is luck because I didn't specifically check the code of the third-party lib.. There is no date-related error so far so I can remove this for now, I guess.
